### PR TITLE
CAMEL-10964: Add an itest for Jetty reverse proxy config with @BeanInject in RouteBuilder

### PR DESCRIPTION
**Pull request by tadayosi**
Actual fix for the issue is provided by CAMEL-10513.

https://issues.apache.org/jira/browse/CAMEL-10964

### CODE REVIEW

File: tests/camel-itest-osgi/src/test/resources/org/apache/camel/itest/cxf/CamelCxfBeanInjectTest.xml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+             http://camel.apache.org/schema/blueprint http://camel.apache.org/schema/blueprint/camel-blueprint.xsd
+             http://camel.apache.org/schema/blueprint/cxf http://camel.apache.org/schema/blueprint/cxf/camel-cxf.xsd
+             http://cxf.apache.org/transports/http-jetty/configuration http://cxf.apache.org/schemas/configuration/http-jetty.xsd
+             http://aries.apache.org/blueprint/xmlns/blueprint-cm/v1.1.0 http://aries.apache.org/schemas/blueprint-cm/blueprint-cm-1.1.0.xsd
+             http://www.osgi.org/xmlns/blueprint/v1.0.0 https://www.osgi.org/xmlns/blueprint/v1.0.0/blueprint.xsd">
+
+  <!-- blueprint property placeholders -->
+  <cm:property-placeholder persistent-id="my-placeholders" update-strategy="reload">
+    <cm:default-properties>
+      <cm:property name="router.address" value="http://localhost:9000/routerEndpoint" />
```
Comment (davsclaus) [inline tests/camel-itest-osgi/src/test/resources/org/apache/camel/itest/cxf/CamelCxfBeanInjectTest.xml]: Can we avoid hardcoded port numbers as CI server sometimes fail on that

Comment (tadayosi) [inline tests/camel-itest-osgi/src/test/resources/org/apache/camel/itest/cxf/CamelCxfBeanInjectTest.xml]: @davsclaus This hardcoded value is in fact overridden in the test case CamelCxfBeanInjectTest. Could you check that?

**Comment by davsclaus** [comment]
The PR has been merged. But sometimes locally I get
```
Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 19.998 sec <<< FAILURE! - in org.apache.camel.itest.cxf.CamelCxfBeanInjectTest
testReverseProxy(org.apache.camel.itest.cxf.CamelCxfBeanInjectTest)  Time elapsed: 19.988 sec  <<< ERROR!
java.net.ConnectException: Connection refused
	at java.net.PlainSocketImpl.socketConnect(Native Method)
	at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:345)
	at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:206)
	at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:188)
	at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392)
	at java.net.Socket.connect(Socket.java:589)
	at sun.net.NetworkClient.doConnect(NetworkClient.java:175)
	at sun.net.www.http.HttpClient.openServer(HttpClient.java:432)
	at sun.net.www.http.HttpClient.openServer(HttpClient.java:527)
	at sun.net.www.http.HttpClient.<init>(HttpClient.java:211)
	at sun.net.www.http.HttpClient.New(HttpClient.java:308)
	at sun.net.www.http.HttpClient.New(HttpClient.java:326)
	at sun.net.www.protocol.http.HttpURLConnection.getNewHttpClient(HttpURLConnection.java:1168)
	at sun.net.www.protocol.http.HttpURLConnection.plainConnect0(HttpURLConnection.java:1147)
	at sun.net.www.protocol.http.HttpURLConnection.plainConnect(HttpURLConnection.java:998)
	at sun.net.www.protocol.http.HttpURLConnection.connect(HttpURLConnection.java:932)
	at sun.net.www.protocol.http.HttpURLConnection.getOutputStream0(HttpURLConnection.java:1282)
	at sun.net.www.protocol.http.HttpURLConnection.getOutputStream(HttpURLConnection.java:1257)
	at org.apache.cxf.transport.http.URLConnectionHTTPConduit$URLConnectionWrappedOutputStream.setupWrappedStream(URLConnectionHTTPConduit.java:257)
	at org.apache.cxf.transport.http.HTTPConduit$WrappedOutputStream.handleHeadersTrustCaching(HTTPConduit.java:1328)
	at org.apache.cxf.transport.http.HTTPConduit$WrappedOutputStream.onFirstWrite(HTTPConduit.java:1288)
	at org.apache.cxf.transport.http.URLConnectionHTTPConduit$URLConnectionWrappedOutputStream.onFirstWrite(URLConnectionHTTPConduit.java:288)
	at org.apache.cxf.io.AbstractWrappedOutputStream.write(AbstractWrappedOutputStream.java:47)
	at org.apache.cxf.io.AbstractThresholdOutputStream.write(AbstractThresholdOutputStream.java:69)
	at org.apache.cxf.transport.http.HTTPConduit$WrappedOutputStream.close(HTTPConduit.java:1341)
	at org.apache.cxf.transport.AbstractConduit.close(AbstractConduit.java:56)
	at org.apache.cxf.transport.http.HTTPConduit.close(HTTPConduit.java:653)
	at org.apache.cxf.interceptor.MessageSenderInterceptor$MessageSenderEndingInterceptor.handleMessage(MessageSenderInterceptor.java:62)
	at org.apache.cxf.phase.PhaseInterceptorChain.doIntercept(PhaseInterceptorChain.java:308)
	at org.apache.cxf.endpoint.ClientImpl.doInvoke(ClientImpl.java:514)
	at org.apache.cxf.endpoint.ClientImpl.invoke(ClientImpl.java:423)
	at org.apache.cxf.endpoint.ClientImpl.invoke(ClientImpl.java:324)
	at org.apache.cxf.endpoint.ClientImpl.invoke(ClientImpl.java:277)
	at org.apache.cxf.frontend.ClientProxy.invokeSync(ClientProxy.java:96)
	at org.apache.cxf.frontend.ClientProxy.invoke(ClientProxy.java:81)
	at com.sun.proxy.$Proxy95.op(Unknown Source)
	at org.apache.camel.itest.cxf.CamelCxfBeanInjectTest.testReverseProxy(CamelCxfBeanInjectTest.java:80)


Results :

Tests in error:
  CamelCxfBeanInjectTest.testReverseProxy:80 » Connect Connection refused

Tests run: 7, Failures: 0, Errors: 1, Skipped: 1
```

**Comment by tadayosi** [comment]
Hi @davsclaus, can you check the 2nd pull req please?

Because it fails only sometimes it's not 100% sure that this one really hit the test issue, but so far the test always passes at my end. I raised the minimum port number used to `30000` here just to be safe, but the effective fix is at `AbstractFeatureTest`.

Thank you!

**Comment by davsclaus** [comment]
Thanks just merged your 2nd commit

**Comment by davsclaus** [comment]
The cxf test now passes on my laptop as well.

**Pull request closed without merging**